brambling/views/payment.py

```diff
@@ -120,3 +124,89 @@ def post(self, request, *args, **kwargs):
         txn.is_confirmed = True if data['Transaction']['Status'] == 'processed' else False
         txn.save()
         return HttpResponse('')
+
+
+class StripeWebhookView(View):
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, *args, **kwargs):
+        return super(StripeWebhookView, self).dispatch(*args, **kwargs)
+
+    def post(self, request):
+        if request.META['CONTENT_TYPE'] != 'application/json':
+            raise Http404('Incorrect content type')
+
+        try:
+            event_data = json.loads(request.body)
+        except ValueError:
+            raise Http404('Webhook failed to decode request body')
+
+        stripe_event_id = event_data.get('id')
+        if not stripe_event_id:
+            raise Http404('Event does not have an id')
+
+        if event_data.get('livemode', False):
+            stripe_prep(LIVE)
+        else:
+            stripe_prep(TEST)
+
+        try:
+            event = stripe.Event.retrieve(stripe_event_id)
+        except stripe.error.InvalidRequestError:
+            raise Http404('Event not found on stripe')
+
+        if event.type != 'charge.refunded':
+            return HttpResponse(status=200)
+
+        with transaction.atomic():
```

Comment:
This following section should probably just use [get_or_create](https://docs.djangoproject.com/en/1.9/ref/models/querysets/#get-or-create)